### PR TITLE
Rector for creating Template class and adding latte {templateType} based on presenter template parameters

### PR DESCRIPTION
**Pull request by lulco**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -56,9 +56,18 @@ parameters:
...
+            paths:
+                - src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php
+                - src/Rector/Class_/TemplateTypeBasedOnPresenterTemplateParametersRector.php
+
+        # PHPStan false positive. There is no method configure on reported line
+        -
+            message: '#Method configure\(\) with passed value object must contain assert to verify passed type#'
+            path: src/Rector/Class_/TemplateTypeBasedOnPresenterTemplateParametersRector.php
 
         # this is the way how it works in rector-src
```
Comment (lulco) [inline phpstan.neon:68]: Should we change it here? https://github.com/rectorphp/rector-src/blob/main/src/NodeFactory/ClassWithPublicPropertiesFactory.php#L28

Comment (TomasVotruba) [reply phpstan.neon:68]: Yes, the array is very crappy to work with. Should be value object instead

Comment (lulco) [reply phpstan.neon:68]: Ok but it will be BC break right? Or it is not a problem?

Comment (TomasVotruba) [reply phpstan.neon:68]: I think the `TemplateTypeBasedOnPresenterTemplateParametersRector` is our internal rule, so we can change anything there.

Also on 0.x BC breaks are allowed.

Comment (lulco) [reply phpstan.neon:68]: I ment to change the ClassWithPublicPropertiesFactory in rector-src. I think I am the only one who uses it in rector-nette but who knows :)

Comment (TomasVotruba) [reply phpstan.neon:68]: Yes :slightly_smiling_face: , go for it :+1:

Comment (lulco) [reply phpstan.neon:68]: Can you please check this PR without the change in rector-src? Currently I am not able to do this refactoring.

Comment (TomasVotruba) [reply phpstan.neon:68]: Sure :+1:

**Comment by TomasVotruba** [comment]
Thank you for this rule :+1:

**Pull request merged**